### PR TITLE
Verify compatibility of provided Test262 directory

### PR DESCRIPTION
**Pull request by jugglinmike**
Future versions of that project may alter the requirements for test
generation. Because the user of this library may provides any arbitrary
version of Test262, it is possible that some future change would cause
this library to emit invalid test material.

Reference the `package.json` file provided by Test262 to determine
compatibility and reject incompatible versions. Because consumers may
only be interested in a subset of tests which are not effected by any
given "breaking change," expose a configuration option named
`acceptVersoin` to disable this restriction.

### CODE REVIEW

**Comment by jugglinmike** [comment]
@leobalter @spectranaut @rwaldron Would you mind taking a look?

File: test/collateral/invalid-version-unsupported/fake-test262/package.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "version": "2.0.0"
+}
```
Comment (leobalter) [inline test/collateral/invalid-version-unsupported/fake-test262/package.json]: nice!

Comment (jugglinmike) [reply test/collateral/invalid-version-unsupported/fake-test262/package.json]: :)

**Comment by leobalter** [comment]
this looks good to land.

**Pull request merged**